### PR TITLE
Add support for artifactory

### PR DESCRIPTION
**Pull request by TheAssassin**
This PR adds support for artifactory as an alternative to transfer.sh for uploading binaries for PRs.

### CODE REVIEW

**Comment by probonopd** [comment]
Isn't https://jfrog.com/artifactory/ the paid version of Bintray? Do we have an Artifactory account to test with? (I'd like to have one)

**Comment by TheAssassin** [comment]
Artifactory is not the paid version of Bintray. It's actually what you would use one stage before pushing into Bintray (it integrates with Bintray even) in a CD scenario, after e.g., having reviewed your binaries.

Artifactory is available in an OSS version (AGPL) with a limited feature set, but its "generic" repository type works fine for simply uploading files and looking at them. It's similar to e.g., [minio](https://www.minio.io/), but since it is designed to be used in CI/CD scenarios, it allow for browsing files anonymously, provides a "normal" HTTP server view, so it should work fine for use with uploadtool.

A demo instance can be found here: https://artifacts.assassinate-you.net/artifactory.

I intend to archive old binaries we generate on my private infrastructure to help with debugging older issues.

Actually, thinking about this PR, I should move the upload code so if artifactory data is available, it will always upload to it, so we can also archive the released binaries.

Another TODO is description support. Still reading in the API docs how to do that.

I already set up AppImageUpdate with support for this instance, so once you merge this PR, we'll upload all files to Artifactory, too.

I'm not 100% fond with it yet (although it's quite flexible even for being some sort of freemium product), but it's the best I could come up with. It won't replace GitHub releases, of course.

**Comment by TheAssassin** [comment]
TL;DR: Please don't merge yet, still WIP.

**Comment by TheAssassin** [comment]
@probonopd I'm finished, please merge.

**Comment by probonopd** [comment]
> I intend to archive old binaries we generate on my private infrastructure

That's of course great, as long as we don't rely on "private cloud" solutions for critical AppImage infrastructure, which I want to have in the "public cloud" (GitHub, GitLab, and the likes).

**Comment by TheAssassin** [comment]
> That's of course great, as long as we don't rely on "private cloud" solutions for critical AppImage infrastructure, which I want to have in the "public cloud" (GitHub, GitLab, and the likes).

... really? Of course we won't "rely" on it. This is all optional.
